ibm/service/vpc/resource_ibm_is_security_group_rule.go

```diff
@@ -679,6 +673,15 @@ func parseIBMISSecurityGroupRuleDictionary(d *schema.ResourceData, tag string, s
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("[ERROR] Error calling asPatch for SecurityGroupRulePatch: %s", err)
 	}
+	if _, ok := d.GetOk("icmp"); ok {
+
+		if parsed.icmpType == -1 {
```

Comment:
Valid values are 0 to 255 how can we compare to "-1"

Reply:
If we check the beginning of the method we initialise with -1, so this code is to handle the scenario of "icmp {}"

func parseIBMISSecurityGroupRuleDictionary

<img width="521" alt="Screenshot 2023-04-20 at 11 43 06 AM" src="https://user-images.githubusercontent.com/127872893/233274056-bcb756ad-1eda-4386-b9c5-23c4161826f4.png">